### PR TITLE
Miscellaneous fixes

### PR DESCRIPTION
**Pull request by davidchappelle**
This pull request contains miscellaneous small fixes for things like build options, etc.

### CODE REVIEW

File: autobahn/wamp_session.ipp [hunk 1]
```diff
@@ -683,7 +682,6 @@ void wamp_session<IStream, OStream>::process_challenge(const wamp_message& messa
             itr = details.find("salt");
             if (itr != details.end()) {
                 // ok we must salt our secret
-                salted = true;
```
Comment (oberstet) [inline autobahn/wamp_session.ipp]: really?

Comment (davidchappelle) [inline autobahn/wamp_session.ipp]: My compiler errors out saying that the variable is only ever assigned and never used. So basically, it exists unnecessarily. If it is actually supposed to be used for something perhaps the original author can comment further?

Comment (oberstet) [inline autobahn/wamp_session.ipp]: Ah, right. Seems to be unused ..

**Comment by oberstet** [comment]
I've done the license change (#62) now (sorry, this broke this PR). I guess https://github.com/leoyanggit (who contributed to this PR) is in agreement too?

**Comment by jpetso** [comment]
Yep, Leo's with Topology, of whose members you've got a blanket agreement.

**Comment by oberstet** [comment]
@davidchappelle the problem is: this PR is sourcing from your master branch .. which is a moving target. A better approach is using feature/bugfix branches. One (throw-away) branch for each PR. This is an established workflow in OSS, and in particular on GitHub.

The changes here are actually 3 things: a) whitespace, b) build options and c) remove unused var.

I'd be fine with 1 PR in this case - but from a branch dedicated to that (other OSS projects have strict policies: 1 thing = 1 branch = 1 PR - eg Twisted wouldn't merge above in 1 PR).

**Comment by davidchappelle** [comment]
Yes we have had some challenges here. Initially this upstream repo was not in a very usable state and needed a lot of work. Unfortunately, uptake of patches was also fairly slow. So we have been basically just maintaining our own "master" branch of autobahn-cpp. As a result, we have our downstream master branch which we our own developers submit pull requests to or, for the sake of simplicity, just push directly to our own master branch.

**Comment by davidchappelle** [comment]
Let me see if I can split these commits up how you would like. After that we can look at how to get the transport abstraction work upstream as well.

**Pull request closed without merging**